runtime/j9vm/java11vmi.c

```diff
@@ -866,19 +932,46 @@ JVM_DefineModule(JNIEnv * env, jobject module, jboolean isOpen, jstring version,
  * 4) Package is not defined for fromModule's class loader
  * 5) Package is not in module fromModule.
  */
+#if JAVA_SPEC_VERSION >= 15
+void JNICALL
+JVM_AddModuleExports(JNIEnv * env, jobject fromModule, jstring packageObj, jobject toModule)
+#else
 void JNICALL
 JVM_AddModuleExports(JNIEnv * env, jobject fromModule, const char *package, jobject toModule)
+#endif /* JAVA_SPEC_VERSION >= 15 */
 {
 	J9VMThread * const currentThread = (J9VMThread*)env;
 	J9JavaVM const * const vm = currentThread->javaVM;
 	J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
+#if JAVA_SPEC_VERSION >= 15
+	const char *package = NULL;
+	PORT_ACCESS_FROM_ENV(env);
+#endif /* JAVA_SPEC_VERSION >= 15 */
 
 	vmFuncs->internalEnterVMFromJNI(currentThread);
 #if defined(CALL_BUNDLED_FUNCTIONS_DIRECTLY)
 	omrthread_monitor_enter(vm->classLoaderModuleAndLocationMutex);
 #else
 	f_monitorEnter(vm->classLoaderModuleAndLocationMutex);
 #endif /* defined(CALL_BUNDLED_FUNCTIONS_DIRECTLY) */
+	
+#if JAVA_SPEC_VERSION >= 15
+	if (NULL != packageObj) {
+		j9object_t stringObject = J9_JNI_UNWRAP_REFERENCE(packageObj);
+		UDATA utfLength = vmFuncs->getStringUTF8Length(currentThread, stringObject) + 1;
+		char* packageName = (char *)j9mem_allocate_memory(utfLength, OMRMEM_CATEGORY_VM);
```

Comment:
similar potential optimization here by keeping a small stack allocated buffer to avoid the malloc

Reply:
Will open a separate issue once this piece of code is merged.